staking-pool/README.md

```diff
@@ -298,3 +298,140 @@ pub fn update_reward_fee_fraction(&mut self, reward_fee_fraction: RewardFeeFract
 /// behalf of this staking pool.
 pub fn vote(&mut self, voting_account_id: AccountId, proposal_id: ProposalId) -> Promise;
 ```
+
+## Migrating from an existing validator or contract
+This provides instructions to migrate your staked validator or a validator contract to a new contract
+
+#### Upgrade to the latest near-shell:
+```bash
+npm install -g near-shell
+```
+#### Set Environment and Login:
+
+##### If not logged into the browser, recover your account with the seed phrase first
+https://wallet.betanet.nearprotocol.com/create/
+
+```bash
+#Set the NEAR environment to the target network (betanet,testnet,mainnet)
+export NEAR_ENV=betanet
+
+near login
+```
+
+#### Unstake and Withdraw:
+```bash
+#If you staked to your validator unstake, there is no withdraw
+near stake nearkat.betanet <staking public key> 0
+
+#If you staked to a contract get the staked balance
+near view my_validator get_account_staked_balance '{"account_id": "user1"}'
+
+#Unsake by copying and pasting the staked balance
+near call my_validator unstake '{"amount": "100000000000000000000000000"}' --accountId user1
+
+#Wait 4 epochs (12 hours) to withdraw and check if balance is available to withdraw
+near view my_validator is_account_unstaked_balance_available '{"account_id": "user1"}'
+
+#If is_account_unstaked_balance_available returns "true" withdraw
+near call my_validator withdraw '{"amount": "100000000000000000000000000"}' --accountId user1
+```
+#### Download new contract with Git:
+```bash
+mkdir staking-pool
+
+cd staking-pool
+
+git clone https://github.com/near/initial-contracts
+
+cd initial-contracts
+
+cd staking-pool
+```
+#### Build contract with Rust (This step is optional since the contract is compiled):
+##### Install Rust:
+```bash
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+
+#Add rust to current shell path
+source $HOME/.cargo/env
+```
+##### Add wasm target to your toolchain:
+```bash
+rustup target add wasm32-unknown-unknown
+```
+##### Build:
+```bash
+./build.sh
+```
+#### Get validator key:
+```bash
+cat ~/.near/betanet/validator_key.json |grep "public_key"
+```
+#### Create a new account to deploy contract to
+- Set my_validator to the name you want publicly displayed
+- --masterAccount is your account you signed up to StakeWars2 with
+```bash
+near create_account my_validator --masterAccount=owner
```

Comment:
This step might fail because it's executed from the `staking-pool/initial-contracts/staking-pool` folder. It might not have `neardev` folder that you used to login.

I suggest to link it here using `ln -s ~/neardev neardev` (assuming you logged in from `~`)

Reply:
This is old information. New `near-shell` stores everything under `~/.near-credentials/`, so disregard this comment.